python/ray/_raylet.pyx

```diff
@@ -2726,7 +2725,14 @@ cdef class GcsClient:
                   nums_reconnect_retry=RayConfig.instance().nums_py_gcs_reconnect_retry(
                   ),
                   cluster_id: str = None):
-        cdef GcsClientOptions gcs_options = GcsClientOptions.from_gcs_address(address)
+        cdef GcsClientOptions gcs_options
+        if cluster_id:
+            gcs_options = GcsClientOptions.create(
+                address, cluster_id, allow_cluster_id_nil=True,
+                fetch_cluster_id_if_nil=True)
+        else:
+            gcs_options = GcsClientOptions.create(
+                address, None, allow_cluster_id_nil=True, fetch_cluster_id_if_nil=True)
```

Comment:
This should be true false?

Reply:
Today it fetches cluster id if it's nil, so True True: https://github.com/ray-project/ray/blob/master/src/ray/gcs/gcs_client/gcs_client.cc#L169